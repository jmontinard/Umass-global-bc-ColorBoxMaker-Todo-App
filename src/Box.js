import React from 'react';

function Box({ id, width, height, backgroundColor, removeBox }) {
  const style = {
    width: `${width}px`,
    height: `${height}px`,
    backgroundColor,
  };

  return (
    <div style={{ display: 'inline-block', margin: '10px' }}>
      <div style={style} />
      <button onClick={() => removeBox(id)}>X</button>
    </div>
  );
}

export default Box;
