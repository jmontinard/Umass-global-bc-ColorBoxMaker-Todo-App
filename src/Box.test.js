import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

test('Box component renders without crashing', () => {
  render(
    <Box
      id="1"
      width="100"
      height="100"
      backgroundColor="red"
      removeBox={() => {}}
    />
  );
});

test('Box component matches snapshot', () => {
  const { asFragment } = render(
    <Box
      id="1"
      width="100"
      height="100"
      backgroundColor="red"
      removeBox={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
