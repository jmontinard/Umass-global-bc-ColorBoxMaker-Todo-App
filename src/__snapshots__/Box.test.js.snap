// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; margin: 10px;"
  >
    <div
      style="width: 100px; height: 100px; background-color: red;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
